### ISSUE TITLE
[java] false positive UnusedAssignment with Lombok @Builder and @NoArgsConstructor

### ISSUE THREAD

**Issue by tmiveda** [pmd/pmd#2817]
**Affects PMD Version:** 
6.27.0

**Description:**
When using PMD together with Lombok's \@Builder and \@NoArgsConstructor the _UnusedAssignment_ warning will apear.

In the example below there is a  _variable2_ that is assigned directly. There is a Constructor marked with \@Builder where the variable is assigned again. 
But there is also a chance to construct the class using the _NoArgsConstructor_. In that case the assignment on class level would apply and the result may be unexpected.

**Code Sample demonstrating the issue:**
```
package net.veda.pmd.test;

import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
class PmdUnused {

	private String variable1;
	private Double variable2 = (double) 0;

	@Builder
	public PmdUnused(String variable1, Double variable2) {
		this.variable1 = variable1;
		this.variable2 = variable2;
	}
}
```

**Steps to reproduce:**
Just check the above code with a ruleset containing _UnusedAssignment_ rule.

**Running PMD through:** *Maven , IntelliJ PMD Plugin*
